firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -321,12 +313,27 @@ public void deleteFirebaseInstallation(
 
         throw new FirebaseInstallationsException(
             "Bad config while trying to delete FID", Status.BAD_CONFIG);
+      } catch (IOException ignored) {
+        retryCount++;
       } finally {
         httpURLConnection.disconnect();
       }
     }
 
-    throw new IOException();
+    throw new FirebaseInstallationsException(
+        "Firebase Installations Service is unavailable. Please try again later.",
+        Status.UNAVAILABLE);
+  }
+
+  private URL getURL(String resourceName) throws FirebaseInstallationsException {
+    try {
+      return new URL(
+          String.format(
+              "https://%s/%s/%s",
+              FIREBASE_INSTALLATIONS_API_DOMAIN, FIREBASE_INSTALLATIONS_API_VERSION, resourceName));
+    } catch (MalformedURLException e) {
+      throw new FirebaseInstallationsException(e.getMessage(), Status.UNAVAILABLE);
```

Comment:
This seems to be a pretty dramatic error.
How about we add some message here, e.g.
"Unexpected internal error while creating request URI."
I don't know.
Feel free to ignore.